components/camel-kafka/src/main/java/org/apache/camel/component/kafka/consumer/CommitManagers.java

```diff
@@ -34,13 +34,7 @@ public static CommitManager createCommitManager(
         }
 
         if (configuration.isAutoCommitEnable()) {
-            if ("async".equals(configuration.getAutoCommitOnStop())) {
-                return new AsyncCommitManager(consumer, kafkaConsumer, threadId, printableTopic);
-            } else if ("sync".equals(configuration.getAutoCommitOnStop())) {
-                return new SyncCommitManager(consumer, kafkaConsumer, threadId, printableTopic);
-            } else if ("none".equals(configuration.getAutoCommitOnStop())) {
-                return new NoopCommitManager(consumer, kafkaConsumer, threadId, printableTopic);
-            }
+            return new AsyncCommitManager(consumer, kafkaConsumer, threadId, printableTopic);
```

Comment:
I don't think it is needed because no matter what the commit manager does, the offset will be handled automatically on consumer close. Wouldn't it be better to return NoopCommitManager here?

Reply:
I think it could be. There's more to it, though.